### ISSUE TITLE
JSON-RPC_API v9 (Leia) seems broken for Files.GetFileDetails

### ISSUE THREAD

**Issue by nomadyow** [xbmc/xbmc#14734]
### Describe the bug

Sending the following JSON worked in Krypton but now fails in Leia.  No documented changes in:

https://kodi.wiki/view/JSON-RPC_API/v9#Files.GetFileDetails
or
https://forum.kodi.tv/showthread.php?tid=324598

Seem to indicate a change in this particular API.

## Actual Behavior
Send this:
{"jsonrpc":"2.0","method":"Files.GetFileDetails","params":{"file":"smb://10.0.0.7/movies/Star Trek (2009).mkv","media":"video","properties":["title"]},"id":1}

returns:

{"error":{"code":-32602,"message":"Invalid params."},"id":1,"jsonrpc":"2.0"}

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Raspberri-Pi
 - [X] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version:
18 beta 4

**Comment by DaveTBlake** [comment]
I see `Files.GetFileDetails` in v18 behaving just like v17, there is no regression.

Note that the "file" param is case sensitive, and the path must match exactly. On Windows this means using "\\" (escaping for JSON) rather than "/".  Hence
```
{"jsonrpc": "2.0", "id": 0, "method":"Files.GetFileDetails", 
"params":{"file":"C:\\TestVideos\\Star Trek (2009).mkv", 
"media":"video", "properties":["title"]}}
```
works, and not "C:/Testvideos/star Trek (2009).mkv". I suspect that is why you see the error return.

What impact Samba has I am unsure and unable to test, hence leaving this issue open until someone can confirm.

**Comment by nomadyow** [comment]
This is the exact JSON I'm using that is throwing the error - including the escaping:
{"jsonrpc":"2.0","method":"Files.GetFileDetails","params":{"file":"smb:\\\\10.0.0.7\\movies\\Star Trek (2009).mkv","media":"video","properties":["title"]},"id":1}

This is the filename that Kodi reports:
smb://10.0.0.7/movies/Star Trek (2009).mkv

I think I have the case sensitivity and escaping set correctly.

**Comment by nomadyow** [comment]
Ahh - I think I figured it out based on your clue.  The filename works fine, but it is not finding it with that path in the MySQL DB because the it was likely stored with a different variation on the path.  I probably changed the way the path was accessed when I setup Leia (I did a fresh install).

I'll leave this open for now, but I will likely close it soon when I've validated that this is the case.
